common/k8-events.go

```diff
@@ -0,0 +1,65 @@
+package common
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+	"k8s.io/client-go/kubernetes"
+)
+
+// K8Event abstracts kubernetes event.
+type K8Event struct {
+	// The object that this event is about.
+	Name string
+	// Namespace where event should be created
+	Namespace string
+	// What action was taken/failed regarding to the Regarding object.
+	Action string
+	// This should be a short, machine understandable string that gives the reason
+	// for the transition into the object's current status.
+	Reason string
+	// Kind of component generating this event
+	Kind string
+	// Name of the controller that emitted this Event
+	ReportingController string
+	// ID of the controller instance
+	ReportingInstance string
+	// Type of this event (Normal, Warning), new types could be added in the future
+	Type string
+	// Map of string keys and values that can be used to organize and categorize
+	// (scope and select) objects.
+	Labels map[string]string
+}
+
+// CreateK8Event returns a kubernetes event object
+func GetK8Event(event *K8Event) *corev1.Event{
+	return &corev1.Event{
+		Reason: event.Reason,
+		Type:   event.Type,
+		Action: event.Action,
+		EventTime: metav1.MicroTime{
+			Time: time.Now(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: event.Namespace,
+			Name:      event.Name + "-" + RandomStringGenerator(),
```

Comment:
can you not just use `GenerateName` here instead?

Reply:
yup, changed it in my local. I'll push some new changes with test cases